Orange/data/table.py

```diff
@@ -796,17 +796,17 @@ def __setitem__(self, key, value):
             if value is None:
                 value = Unknown
 
-            if not isinstance(value, Real) and \
+            if not isinstance(value, (Real, np.ndarray)) and \
                     (len(attr_cols) or len(class_cols)):
                 raise TypeError(
                     "Ordinary attributes can only have primitive values")
             if len(attr_cols):
-                if len(attr_cols) == 1:
+                if sp.issparse(self.X) and len(attr_cols) == 1:
```

Comment:
The original check has been here for a long time :)

I have removed the if completely.